airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_18_004__BackfillActorDefinitionWorkspaceGrant.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.instance.configs.migrations;
+
+import java.util.UUID;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: update migration description in the class name
+public class V0_40_18_004__BackfillActorDefinitionWorkspaceGrant extends BaseJavaMigration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(
+      V0_40_18_004__BackfillActorDefinitionWorkspaceGrant.class);
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
+    final DSLContext ctx = DSL.using(context.getConnection());
+
+    var customActorDefinitionIds = ctx.fetch("SELECT id FROM actor_definition WHERE public is false and tombstone is false;");
+    var existingWorkspaces = ctx.fetch("SELECT id FROM WORKSPACE where tombstone is false;");
+
+    // Update for all custom connectors - set custom field to true;
+    ctx.execute("UPDATE actor_definition"
+        + " SET custom = true "
+        + " WHERE public is false and tombstone is false;");
+
+    for (final var customActorDefinitionIdRecord : customActorDefinitionIds) {
+      for (final var existingWorkspaceRecord : existingWorkspaces) {
```

Comment:
seems like something you could try doing with a sql for loop if you're feeling adventurous 😄 

Unfortunately however it looks like the open source version of jooq doesn't support loops https://www.jooq.org/doc/latest/manual/sql-building/procedural-statements/procedural-loop/ so it's probably not worth trying to get this work (we'd have to hand write sql for postgres specifically, which doesn't sound good).

Reply:
Good to know there is a LOOP function in SQL :) 

I think since this is just one time migration I'll stick to current solution since it has been tested